packages/notifier/src/types-ambient.js

```diff
@@ -101,6 +101,45 @@
 
 // /////////////////////////////////////////////////////////////////////////////
 
+/**
+ * @typedef {'mandatory' | 'opportunistic' | 'ignored'} DurablePublishKitValueDurability
+ *
+ * Durability configures constraints on non-terminal values that can be
+ * published to a durable publish kit (terminal values sent to finish or fail
+ * must always be durable).
+ * - 'mandatory' means that each value must be durable, so it can be restored
+ *   on upgrade.
+ * - 'opportunistic' means that a durable value is persisted for post-upgrade
```

Comment:
do we really have a use case for this? sounds confusing, to have the value get mysteriously replaced by `undefined` after an ugprade

Reply:
@dtribble requested the ability to publish non-durable values to a durable publish kit, and this is the sketch of what it would look like (but not that the implementation does not actually support it and will throw upon an attempt to use it).